pypeit/edgetrace.py

```diff
@@ -4284,10 +4284,30 @@ def maskdesign_matching(self, debug=False):
             return
 
         # `traceimg` must have knowledge of the flat frame that built it
-        self.maskfile = self.traceimg.files[0]
+        wcs_file = None
+        if self.par['maskdesign_filename'] is not None: 
+            fpath = os.path.dirname(self.traceimg.files[0])
+            if isinstance(self.par['maskdesign_filename'], str):
+                self.maskfile = self.par['maskdesign_filename']
+                # Add path?
+                if not os.path.isfile(self.maskfile):
+                    self.maskfile = os.path.join(fpath, self.maskfile)
+            elif isinstance(self.par['maskdesign_filename'], list):
+                self.maskfile = self.par['maskdesign_filename'][0]
+                wcs_file = self.par['maskdesign_filename'][1]
+                # Add path?
+                if not os.path.isfile(self.maskfile):
+                    self.maskfile = os.path.join(fpath, self.maskfile)
+                if not os.path.isfile(wcs_file):
+                    wcs_file = os.path.join(fpath, wcs_file)
+        else:
+            self.maskfile = self.traceimg.files[0] 
```

Comment:
Two things:

1. I think most/all of this should now go in the spectrograph code.  Here, I think the code should be something like:
```
maskfiles = self.traceimg.file[0] if self.par['maskdesign_filename'] is None else self.par['maskdesign_filename']
omodel_bspat, omodel_tspat, sortindex, self.slitmask = \
        self.spectrograph.get_maskdef_slitedges(ccdnum=self.traceimg.detector.det, binning=self.traceimg.detector.binning, filename=maskfiles, debug=debug)
```

2. Currently, the `maskfile` is saved to `self` because it is passed to the `SlitTraceSet` object mostly for documentation/logging.  Should we now be doing the same for the "wcs_file"; should we save the `maskfiles` from above instead?

Reply:
good call

Reply:
Item 1 is done.

Item 2 updated, but not wcs_file